### PR TITLE
fix(sdk): align sdk with platform modes

### PR DESCRIPTION
**Pull request by strantalis**
Resolves: https://github.com/opentdf/platform/issues/1360

### CODE REVIEW

File: sdk/options.go [hunk 1]
```diff
@@ -22,18 +22,19 @@ type config struct {
 	tokenExchange           *oauth.TokenExchangeInfo
 	tokenEndpoint           string
 	scopes                  []string
-	policyConn              *grpc.ClientConn
-	authorizationConn       *grpc.ClientConn
-	entityresolutionConn    *grpc.ClientConn
+	policyConn              *grpc.ClientConn // Deprecated
```
Comment (jakedoublev) [inline sdk/options.go:25]: Question: does the deprecation notice work if you in-line it? I couldn't find a clear answer in the docs and it looks like all examples on the first two pages of code searches in GH are the line above: https://github.com/search?q=%22%2F%2F+deprecated%22+path%3A*.go&type=code&p=1

Comment (strantalis) [reply sdk/options.go:25]: I am honestly not sure. It was more for an internal reminder. I am really not sure if we should deprecate them or just rip them out.

Comment (jrschumacher) [reply sdk/options.go:25]: At this point, I think we need to defecate since we know there's external use.

File: sdk/options.go [hunk 2]
```diff
@@ -99,18 +100,21 @@ func withCustomAccessTokenSource(a auth.AccessTokenSource) Option {
 	}
 }
 
+// Deprecated: Use WithCustomCoreConnection instead
 func WithCustomPolicyConnection(conn *grpc.ClientConn) Option {
```
Comment (jrschumacher) [inline sdk/options.go:104]: Should we just reroute this to the customcoreconnection?

Comment (strantalis) [reply sdk/options.go:104]: I am thinking about that. Does that get confusing and will that break anyone currently using multiple custom connections ?

Comment (jrschumacher) [reply sdk/options.go:104]: Yes, but if they are deprecated then the IDE should complain. A deployment may fail, but we can tell them that we no longer support that case. I'll leave it up to you.

Comment (strantalis) [reply sdk/options.go:100]: Updated

File: sdk/options.go (hunk 1 above)
Comment (strantalis) [reply sdk/options.go]: I just removed this internal fields and now just set them to coreConn

**Review by jrschumacher — APPROVED** [review]

**Review by pflynn-virtru — APPROVED** [review]

**Pull request merged**